### PR TITLE
Prepare repo for prow onboard

### PR DESCRIPTION
**Pull request by jlpadilla**
**Related Issue:**  open-cluster-management/backlog#9992

### Description of changes
- Add `Makefile.prow`
- Add `Dockerfile.prow`
- Update Dockerfile to be consistent with Dockerfile.prow
- Cleanup old Dockerfiles for Power and Z images.
- Upgrade to Go 1.16

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -33,15 +40,17 @@ LABEL org.label-schema.vendor="Red Hat" \
       io.k8s.description="$IMAGE_DESCRIPTION" \
       io.openshift.tags="$IMAGE_OPENSHIFT_TAGS"
 
-RUN microdnf install ca-certificates vi --nodocs &&\
+RUN microdnf update &&\
+    microdnf install ca-certificates vi --nodocs &&\
```
Comment (mqhackett) [inline Dockerfile:44]: 👍 for adding update.

File: Makefile.prow [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Copyright Contributors to the Open Cluster Management project
+
+-include /opt/build-harness/Makefile.prow
+
+.PHONY: build
+build:
+	CGO_ENABLED=0 go build -o $(BINDIR)/search-aggregator ./
```
Comment (mqhackett) [inline Makefile.prow:7]:
```suggestion
	CGO_ENABLED=0 go build -o $(BINDIR)/search-aggregator ./
		
.PHONY: unit-test
unit-test:
	DEPLOYED_IN_HUB=true go test ./... -v -coverprofile cover.out
```

Comment (mqhackett) [reply Makefile.prow:7]: Can we try to add unit-test to the makefile in this PR? It already looks good to merge. I don't want to make the PR too big, but it'd be nice to have this all in one spot

Comment (jlpadilla) [reply Makefile.prow]: Updated, also removed the `build` target because we don't use it from prow

**Review by mqhackett — APPROVED** [review]

**Pull request merged**